### DISCUSSION TITLE
[very important] [easy to fix] [unforeseen consequences] ColorExtensions.Multiply doesn't work correctly with negative numbers

### DISCUSSION THREAD

**Discussion by SomeRandomNoobKekeke** [FakeFishGames/Barotrauma#14150] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

This for example makes halo too bright during jovian storm flashes in the end mission
Same happens if you make a biome with bright ambience

https://github.com/FakeFishGames/Barotrauma/blob/4a63dacbcee8e3829a44e9c16a5335704a979a4d/Barotrauma/BarotraumaClient/ClientSource/Map/Lights/LightManager.cs#L533

It ignores the sign, if `0.3f - ambientBrightness` is negative here then halo will be bright

---
I mean, i guess it's intended to be consistent with "Color * float" which always gives you transparent color if float is negative 

![20240617161059_1](https://github.com/FakeFishGames/Barotrauma/assets/122838333/14b9b27a-64f3-4888-80d7-6b177119c5e5)

![Screenshot_6](https://github.com/FakeFishGames/Barotrauma/assets/122838333/753c9e95-4694-49ca-8080-0847baad5d2b)

### Reproduction steps
`DebugConsole.NewMessage($"Color.White.Multiply(-0.1f) = {Color.White.Multiply(-0.1f)}", Color.Cyan);`

OR

Go to end middle, observe

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Single player

### Version

v1.4.6.0 (Blood in the Water Update, hotfix 2)

### Which operating system did you encounter this bug on?

Windows

**Comment by Regalis11** [comment]
Addressed in https://github.com/FakeFishGames/Barotrauma-development/commit/4eaf6f6c8782e3cd18a06ab73479df87047875a0 by clamping the value between 0 and 255.
